kafka-streams-partitioners/weighted-group-partitioner/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/KeyHashPartitioner.java

```diff
@@ -0,0 +1,18 @@
+package org.hypertrace.core.kafkastreams.framework.partitioner;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+
+public class KeyHashPartitioner<K, V> implements StreamPartitioner<K, V> {
+  private final AtomicInteger currentPartition = new AtomicInteger();
+
+  @Override
+  public Integer partition(String topic, K key, V value, int numPartitions) {
+    int hashcode = 0;
+    if (key != null) {
+      hashcode = key.hashCode();
+    }
+    return Utils.toPositive(hashcode) % numPartitions;
```

Comment:
If, the key is null, this will always return `partition 0`. Is this expected?

Reply:
Based on the test case below, if the key is null, it will always go to the default group.

Reply:
This is just a delegate partitioner and it is used in conjunction with group partitioner. The return value of the delegate partitioner is used as offset and added to the startIndex derived in group partitioner.

Reply:
Can we add the same in the class level documentation

Reply:
Relevant doc here
https://github.com/hypertrace/kafka-streams-framework/blob/main/kafka-streams-partitioners/weighted-group-partitioner/src/main/java/org/hypertrace/core/kafkastreams/framework/partitioner/WeightedGroupPartitioner.java#L14

Reply:
@laxmanchekka The above comment is fine, but I think, we should add a comment here as well at the class level that this is a delegate partitioner and used in conjunction with WeightedGroupPartioner for the future context if anyone reads this class independently.

Reply:
Talked offline about this.

Reply:
This by design is an independent partitioner and can be used standalone as well.
And we have multiple such variants: roundrobin, keyhash, valuehasg, fieldvalue etc.
So, I don't think this documentation is relevant here.

I will go ahead and add everywhere if you think otherwise.